.github/workflows/tests.yml

```diff
@@ -265,33 +265,52 @@ jobs:
       - run: make slim-builds
       - run: make test-integration-splunk
 
-  test-integration-kubernetes:
-    name: Integration - Linux, Kubernetes, flaky
-    # This is an experimental test. Allow it to fail without failing the whole
-    # workflow, but keep it executing on every build to gather stats.
-    continue-on-error: true
+  test-e2e-kubernetes:
+    name: E2E - K8s ${{ matrix.kubernetes_version }} / ${{ matrix.container_runtime }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        kubernetes:
-          - v1.18.2
-          - v1.17.5
-          - v1.16.9
-          - v1.15.11
-          - v1.14.10
+        minikube_version:
+          - 'v1.11.0' # https://github.com/kubernetes/minikube/issues/8799
+        kubernetes_version:
+          - 'v1.18.6'
+          - 'v1.17.9'
+          - 'v1.16.13'
+          - 'v1.15.12'
```

Comment:
Why do we test these versions?

Reply:
According to the RFC, we test all versions from MSKV to the latest released version. MSKV is 1.15.

Reply:
We don't really have to test 1.14, but we do it cause people asked if we'll support it, and our code works for 1.14 too. We might want to reduce MSKV to 1.14.

Reply:
So over time as k8s releases grow, so will this list?  Meaning more and more of our CI jobs will be just k8s tests?

Reply:
I'm afraid so, at least that's the plan. We'll probably bump MSKV along. K8s unfortunately have plenty of edge cases, and we'd better test as many variants as we can.
The good thing is, though, it's quite easy to maintain. I'll improve this a little bit later so we get alerted to add new k8s versions.